File: vortexdbg-api/src/main/java/com/vortexdbg/memory/SvcMemory.java
package com.vortexdbg.memory;

import com.vortexdbg.Svc;
import com.vortexdbg.pointer.VortexdbgPointer;

public interface SvcMemory extends StackMemory {

    VortexdbgPointer allocate(int size, String label);

    VortexdbgPointer allocateSymbolName(String name);

    VortexdbgPointer registerSvc(Svc svc);

    Svc getSvc(int svcNumber);

    MemRegion findRegion(long addr);

    long getBase();
    int getSize();

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/memory/SvcMemory.kt
package com.vortexdbg.memory

import com.vortexdbg.Svc
import com.vortexdbg.pointer.VortexdbgPointer

interface SvcMemory : StackMemory {

    fun allocate(size: Int, label: String): VortexdbgPointer

    fun allocateSymbolName(name: String): VortexdbgPointer

    fun registerSvc(svc: Svc): VortexdbgPointer

    fun getSvc(svcNumber: Int): Svc?

    fun findRegion(addr: Long): MemRegion?

    fun getBase(): Long
    fun getSize(): Int

}
